### DISCUSSION TITLE
Fetch data for layout on language change.

### DISCUSSION THREAD

**Discussion by lazaroofarrill** [nuxt/nuxt#9787] [Questions (Nuxt 2)] [answered]
I've been reading the docs and I still I haven't been able to figure out, how to call the fetch hook for my layout on language change. When I change my language my layout component won't rerender. I read in that the fetch key was supposed to work with that but I have no idea how to use it. Any help would be much appreciated.

**Comment by kissu** [comment, marked as answer]
As said in [the documentation](https://nuxtjs.org/docs/2.x/features/data-fetching#the-fetch-hook)
> In addition to fetch being called by Nuxt, you can manually call fetch in your component (to e.g. reload its async data) by calling this.$fetch()

So, when the locale changes you can call `await this.$fetch()` to reload the `fetch()` hook.

**Reply by lazaroofarrill** [reply]
but how do i detect the locale change if it is triggered in a different component than the one I'm calling fetch

**Reply by jonybekov** [reply]
You can use watcher to detect locale change. So inside your component...
```
...
watch: {
  ['$i18n.locale']() {
     this.$fetch()
  }
}
```
Or
If you are using nuxt-i18n, you can register a callback when your component is mounted

```
mounted() {
  this.$i18n.onLanguageSwitched = (oldLocale, newLocale) => this.$fetch()
}
```

Hope it helps.

**Reply by lazaroofarrill** [reply]
thanks a lot
